### PR TITLE
Initial Haiku support

### PR DESCRIPTION
**Pull request by miqlas**

### CODE REVIEW

**Comment by sfan5** [comment]
@Megaf

File: src/CMakeLists.txt [hunk 1]
```diff
@@ -329,9 +337,11 @@ else()
 	endif(APPLE)
 
 	# This way Xxf86vm is found on OpenBSD too
```
Comment (sfan5) [inline src/CMakeLists.txt]: this comment should be moved inside the `if()` too

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -336,8 +336,8 @@ else()
 		endif(HAVE_LIBRT)
 	endif(APPLE)
 
-	# This way Xxf86vm is found on OpenBSD too
 	if(NOT HAIKU)
+	# This way Xxf86vm is found on OpenBSD too
```
Comment (sfan5) [inline src/CMakeLists.txt]: now you just need to indent it ;)

File: src/client/renderingengine.cpp [hunk 3]
```diff
@@ -34,7 +34,8 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "inputhandler.h"
 #include "gettext.h"
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) && !defined(SERVER) && !defined(__HAIKU__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) \
+	&& !defined(SERVER)	&& !defined(__HAIKU__)
```
Comment (sfan5) [inline src/client/renderingengine.cpp]: the tab in the middle should be a space

**Comment by nerzhul** [comment]
should we support a such OS ?

**Comment by AntonBoch1244** [comment]
@nerzhul should not, but why not to try.
Haiku is a BeOS but more open then BeOS.

**Comment by sfan5** [comment]
the linter still has a complaint about your syntax: https://travis-ci.org/minetest/minetest/jobs/294459402#L504

**Comment by rubenwardy** [comment]
> should we support a such OS ?

Not officially support, ie: no builds or ci tests maintained by us, but this pr is fine

File: src/client/renderingengine.cpp [hunk 4]
```diff
@@ -34,8 +34,8 @@ with this program; if not, write to the Free Software Foundation, Inc.,
 #include "inputhandler.h"
 #include "gettext.h"
 
-#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) \
-	&& !defined(SERVER) && !defined(__HAIKU__)
+#if !defined(_WIN32) && !defined(__APPLE__) && !defined(__ANDROID__) &&
```
Comment (sfan5) [inline src/client/renderingengine.cpp]: still need the `\` on this line

**Pull request merged**